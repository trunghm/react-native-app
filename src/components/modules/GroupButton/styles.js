import EStyleSheet from "react-native-extended-stylesheet";
import GlobalStore from "../../../themes/GlobalStore";

const styles = () => EStyleSheet.create({
  container: {
    width: '100%',
    flexDirection:"row",
    height: 50,
  },
  active:{
    backgroundColor:GlobalStore.color.primary,
  },
  deactive:{
    backgroundColor:GlobalStore.color.buttonBgDisable,
  },
  text: {
    fontSize: 14,
    color: GlobalStore.color.buttonTitle,
    fontWeight: 'bold',
  },

  buttonWrapper:{
    flex: 1, alignItems: "center", justifyContent: "center"
  }
});

export default styles;
