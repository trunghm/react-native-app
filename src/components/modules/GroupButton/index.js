import React, {Component} from 'react';
import styles from "./styles";
import {
  StyleSheet,
  Text,
  View,
  TouchableOpacity
} from 'react-native';
import {Button} from "../../commons";

export default function GroupButton({index, data = [], onSelect}) {
  return (
    <View style={styles().container}>
      {
        data.map((title, _index) => {
          const status = index === _index ? "active" : "deactive";
          return  <Button  key={_index} text={title} disabled={status === "active" ? false : true} full onPress={() => {
            onSelect(_index);
          }}/>
        })
      }
    </View>);
}

export class Document {

  static key = "GroupButton"

  static render = GroupButton

  static demo = {
    selection :[],
    options :[],
    input: [{name: "index", type :"num"},{name:"data", type : "array"}]
  }
}


