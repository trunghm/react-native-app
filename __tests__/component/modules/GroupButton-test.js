/**
 * @format
 */

import 'react-native';
import React from 'react';
import GroupButton from '../../../src/components/modules/GroupButton';

// Note: test renderer must be required after react-native.
import { mount } from 'enzyme';
import EStyleSheet from 'react-native-extended-stylesheet';
import renderer from "react-test-renderer";



const originalError = console.error;

beforeAll(() => {
  console.error = jest.fn();
  EStyleSheet.build({
  });
});

afterAll(() => {
  console.error = originalError;
});

describe('Testing  GroupButton component', () => {

  it('renders snapshot as expected', () => {
    const wrapper = renderer.create(
      <GroupButton data={["button1", "button2"] } index ={1} />
    );

    expect(wrapper).toMatchSnapshot();

  });

  it('renders as expected', () => {
    const wrapper = mount(
      <GroupButton data={["button1", "button2"] } index ={1} />
    );

    expect(wrapper).toMatchSnapshot();
    const button1 = wrapper
      .findWhere((w) => w.prop('text') === 'button1')
      .first();
    const button2 = wrapper
      .findWhere((w) => w.prop('text') === 'button2')
      .first();
    expect(button1.prop('disabled') ).toEqual(true);
    expect(button2.prop('disabled') ).toEqual(false);

  });

});
