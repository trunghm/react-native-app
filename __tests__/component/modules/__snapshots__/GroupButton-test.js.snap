// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing  GroupButton component renders as expected 1`] = `ReactWrapper {}`;

exports[`Testing  GroupButton component renders snapshot as expected 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "maxHeight": 50,
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#d2d2d2",
        "flex": 1,
        "justifyContent": "center",
        "maxHeight": 50,
        "minHeight": 50,
        "opacity": 1,
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 14,
          "fontWeight": "bold",
        }
      }
    >
      BUTTON1
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#1496bb",
        "flex": 1,
        "justifyContent": "center",
        "maxHeight": 50,
        "minHeight": 50,
        "opacity": 1,
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 14,
          "fontWeight": "bold",
        }
      }
    >
      BUTTON2
    </Text>
  </View>
</View>
`;
